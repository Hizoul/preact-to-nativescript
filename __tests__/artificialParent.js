import * as preactToNativeScript from "../index"

const h = preactToNativeScript.Preact.h

test("Basic Render", () => {
  const rendered = preactToNativeScript.render(h("page"))
  const cssFileString = "mycssfile"
  let touchedString = "unotuched"
  expect(cssFileString).not.toBe(touchedString)
  const obj = {
    text: "Hi",
    cssFile: "mycssfile",
    addCssFile: (a) => {
      touchedString = a
    }
  }
  expect(rendered).toMatchSnapshot("after render")
  rendered.parentNode.appendChild(obj)
  expect(cssFileString).toBe(touchedString)
  expect(rendered).toMatchSnapshot("after append child")
  rendered.parentNode.removeChild(obj)
  expect(rendered).toMatchSnapshot("after remove child")
  rendered.parentNode.remove()
  expect(rendered).toMatchSnapshot("after parent remove")
})
