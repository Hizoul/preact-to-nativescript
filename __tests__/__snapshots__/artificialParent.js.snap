// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Render: after append child 1`] = `
Page {
  "__n": "page",
  "__preactattr_": Object {},
  "_eventListener": Array [],
  "attributes": Array [],
  "childNodes": Array [],
  "firstLoad": true,
  "loaded": false,
  "mergeInto": undefined,
  "nodeName": "PAGE",
  "nodeType": 1,
  "parentNode": Object {
    "_removeView": [Function],
    "appendChild": [Function],
    "attributes": Array [],
    "childNodes": Array [
      [Circular],
      Object {
        "addCssFile": [Function],
        "cssFile": "mycssfile",
        "mergeInto": undefined,
        "parentNode": [Circular],
        "renderedComponent": VNode {
          "attributes": undefined,
          "children": Array [],
          "key": undefined,
          "nodeName": "page",
        },
        "text": "Hi",
      },
    ],
    "mergeInto": undefined,
    "nodeName": "artificalParent",
    "nodeType": "artificalParent",
    "remove": [Function],
    "removeChild": [Function],
    "renderedComponent": VNode {
      "attributes": undefined,
      "children": Array [],
      "key": undefined,
      "nodeName": "page",
    },
    "renderedInto": null,
  },
  "renderedComponent": VNode {
    "attributes": undefined,
    "children": Array [],
    "key": undefined,
    "nodeName": "page",
  },
  "renderedInto": [Circular],
  "set": [Function],
}
`;

exports[`Basic Render: after parent remove 1`] = `
Page {
  "__n": "page",
  "__preactattr_": Object {},
  "_eventListener": Array [],
  "attributes": Array [],
  "childNodes": Array [],
  "firstLoad": true,
  "loaded": false,
  "mergeInto": undefined,
  "nodeName": "PAGE",
  "nodeType": 1,
  "parentNode": Object {
    "_removeView": [Function],
    "appendChild": [Function],
    "attributes": Array [],
    "childNodes": Array [
      [Circular],
    ],
    "mergeInto": undefined,
    "nodeName": "artificalParent",
    "nodeType": "artificalParent",
    "remove": [Function],
    "removeChild": [Function],
    "renderedComponent": VNode {
      "attributes": undefined,
      "children": Array [],
      "key": undefined,
      "nodeName": "page",
    },
    "renderedInto": null,
  },
  "renderedComponent": VNode {
    "attributes": undefined,
    "children": Array [],
    "key": undefined,
    "nodeName": "page",
  },
  "renderedInto": [Circular],
  "set": [Function],
}
`;

exports[`Basic Render: after remove child 1`] = `
Page {
  "__n": "page",
  "__preactattr_": Object {},
  "_eventListener": Array [],
  "attributes": Array [],
  "childNodes": Array [],
  "firstLoad": true,
  "loaded": false,
  "mergeInto": undefined,
  "nodeName": "PAGE",
  "nodeType": 1,
  "parentNode": Object {
    "_removeView": [Function],
    "appendChild": [Function],
    "attributes": Array [],
    "childNodes": Array [
      [Circular],
    ],
    "mergeInto": undefined,
    "nodeName": "artificalParent",
    "nodeType": "artificalParent",
    "remove": [Function],
    "removeChild": [Function],
    "renderedComponent": VNode {
      "attributes": undefined,
      "children": Array [],
      "key": undefined,
      "nodeName": "page",
    },
    "renderedInto": null,
  },
  "renderedComponent": VNode {
    "attributes": undefined,
    "children": Array [],
    "key": undefined,
    "nodeName": "page",
  },
  "renderedInto": [Circular],
  "set": [Function],
}
`;

exports[`Basic Render: after render 1`] = `
Page {
  "__n": "page",
  "__preactattr_": Object {},
  "_eventListener": Array [],
  "attributes": Array [],
  "childNodes": Array [],
  "firstLoad": true,
  "loaded": false,
  "mergeInto": undefined,
  "nodeName": "PAGE",
  "nodeType": 1,
  "parentNode": Object {
    "_removeView": [Function],
    "appendChild": [Function],
    "attributes": Array [],
    "childNodes": Array [
      [Circular],
    ],
    "mergeInto": undefined,
    "nodeName": "artificalParent",
    "nodeType": "artificalParent",
    "remove": [Function],
    "removeChild": [Function],
    "renderedComponent": VNode {
      "attributes": undefined,
      "children": Array [],
      "key": undefined,
      "nodeName": "page",
    },
    "renderedInto": null,
  },
  "renderedComponent": VNode {
    "attributes": undefined,
    "children": Array [],
    "key": undefined,
    "nodeName": "page",
  },
  "renderedInto": [Circular],
  "set": [Function],
}
`;
